src/lib/import-export/import/handlers/backup-handler-wpress.ts

```diff
@@ -0,0 +1,219 @@
+
+	let totalBytesToRead = header.size;
+	while ( totalBytesToRead > 0 ) {
+		let bytesToRead = 512;
+		if ( bytesToRead > totalBytesToRead ) {
+			bytesToRead = totalBytesToRead;
+		}
+
+		if ( bytesToRead === 0 ) {
+			break;
+		}
+
+		const buffer = Buffer.alloc( bytesToRead );
+		const data = await fd.read( buffer, 0, bytesToRead, null );
+		outputStream.write( buffer );
+
+		totalBytesToRead -= data.bytesRead;
+	}
+
+	outputStream.close();
+}
+
+export class BackupHandlerWpress extends EventEmitter implements BackupHandler {
+	private bytesRead: number;
+	private eof: Buffer;
+
+	constructor() {
+		super();
+		this.bytesRead = 0;
+		this.eof = Buffer.alloc( HEADER_SIZE, '\0' );
+	}
+
+	/**
+	 * Lists all files in a .wpress backup file by reading the headers sequentially.
+	 *
+	 * It opens the .wpress file, reads each header to get the file names, and stores them in an array.
+	 * The function continues reading headers until it reaches the end of the file.
+	 *
+	 * @param {BackupArchiveInfo} file - The backup archive information, including the file path.
+	 * @returns {Promise<string[]>} - A promise that resolves to an array of file names.
+	 */
+	async listFiles( file: BackupArchiveInfo ): Promise< string[] > {
+		const fileNames: string[] = [];
+
+		if ( ! fs.existsSync( file.path ) ) {
+			throw new Error( `Input file at location "${ file.path }" could not be found.` );
+		}
+
+		const inputFile = await fs.promises.open( file.path, 'r' );
+
+		// Read all of the headers and file data into memory.
+		try {
+			let header;
+			do {
+				header = await readHeader( inputFile );
+				if ( header ) {
+					fileNames.push( path.join( header.prefix, header.name ) );
+					await inputFile.read( Buffer.alloc( header.size ), 0, header.size, null );
+				}
+			} while ( header );
+		} finally {
+			await inputFile.close();
+		}
+
+		return fileNames;
+	}
+
+	/**
+	 * Extracts files from a .wpress backup file into a specified extraction directory.
+	 *
+	 * @param {BackupArchiveInfo} file - The backup archive information, including the file path.
+	 * @param {string} extractionDirectory - The directory where the files will be extracted.
+	 * @returns {Promise<void>} - A promise that resolves when the extraction is complete.
+	 */
+	async extractFiles( file: BackupArchiveInfo, extractionDirectory: string ): Promise< void > {
+		return new Promise( ( resolve, reject ) => {
+			( async () => {
+				try {
+					if ( ! fs.existsSync( file.path ) ) {
+						throw new Error( `Input file at location "${ file.path }" could not be found.` );
+					}
+
+					fse.emptyDirSync( extractionDirectory );
+
+					const inputFile = await fs.promises.open( file.path, 'r' );
+
+					let offset = 0;
+
+					let header;
+					while ( ( header = await readHeader( inputFile ) ) !== null ) {
+						if ( ! header ) {
+							break;
+						}
+
+						await readBlockToFile( inputFile, header, extractionDirectory );
+						offset = offset + HEADER_SIZE + header.size;
```

Comment:
Seems `offset` is not used so we could consider removing it.

Reply:
Wow! great catch! Removed: https://github.com/Automattic/studio/pull/497/commits/e16f25069f0aaa8f37778da22012bc0a3ccc78aa